internal/driver/macmapper.go

```diff
@@ -0,0 +1,114 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+//
+// Copyright (C) 2022 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package driver
+
+import (
+	"fmt"
+	sdk "github.com/edgexfoundry/device-sdk-go/v2/pkg/service"
+	"net"
+	"strings"
+	"sync"
+
+)
+
+type MACAddressMapper struct {
+	// credsMu is for locking access to the credsMap
+	credsMu sync.RWMutex
+	// credsMap is a map between mac address to secretPath
+	credsMap map[string]string
+}
+
+func NewMACAddressMapper() *MACAddressMapper {
+	return &MACAddressMapper{
+		credsMap: make(map[string]string),
+	}
+}
+
+// UpdateMappings takes the raw map of secret path to csv list of mac addresses and
+// inverts it into a quick lookup map of mac address to secret path.
+func (m *MACAddressMapper) UpdateMappings(raw map[string]string) {
+	m.credsMu.Lock()
+	defer m.credsMu.Unlock()
+
+	lc := sdk.RunningService().LoggingClient
+
+	credsMap := make(map[string]string)
+	for secretPath, macs := range raw {
+		if _, err := tryGetCredentials(secretPath); err != nil {
```

Comment:
This should probably just use the first part of `tryGetCredentials` which queries the `SecretProvider` directly. In the future it could make use of https://github.com/edgexfoundry/go-mod-bootstrap/issues/350 if implemented.

```suggestion
		if _, err := sdk.RunningService().SecretProvider.GetSecret(secretPath, UsernameKey, PasswordKey, AuthModeKey); err != nil {
```